frontend/template.yaml

```diff
@@ -0,0 +1,160 @@
+
+Mappings:
+  StageDetails:
+    dev:
+      Subdomain: admin-dev
+    stg:
+      Subdomain: admin-stg
+    prod:
+      Subdomain: admin
+  RegionMap:
+    us-east-1:
+      WebsiteEndpoint: s3-website-us-east-1.amazonaws.com
+    us-west-1:
+      WebsiteEndpoint: s3-website-us-west-1.amazonaws.com
+    us-west-2:
+      WebsiteEndpoint: s3-website-us-west-2.amazonaws.com
+    eu-west-1:
+      WebsiteEndpoint: s3-website-eu-west-1.amazonaws.com
+    ap-southeast-1:
+      WebsiteEndpoint: s3-website-ap-southeast-1.amazonaws.com
+    ap-southeast-2:
+      WebsiteEndpoint: s3-website-ap-southeast-2.amazonaws.com
+    ap-northeast-1:
+      WebsiteEndpoint: s3-website-ap-northeast-1.amazonaws.com
+    sa-east-1:
+      WebsiteEndpoint: s3-website-sa-east-1.amazonaws.com
+
+Resources:
+  WebsiteBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub
+        - cadasta-arcgis-admin-${Stage}
+        - Stage: !Ref Stage
+      AccessControl: PublicRead
+      WebsiteConfiguration:
+        IndexDocument: index.html
+
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      PolicyDocument:
+        Id: S3WebsiteServingPolicy
+        Version: 2012-10-17
+        Statement:
+          - Sid: PublicReadForGetBucketObjects
+            Effect: Allow
+            Principal: '*'
+            Action: s3:GetObject
+            Resource: !Sub arn:aws:s3:::${WebsiteBucket}/*
+      Bucket: !Ref WebsiteBucket
+
+  LogBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub ${WebsiteBucket}-logs
+      LifecycleConfiguration:
+        Rules:
+        - Id: DeleteIn3Months
+          Status: Enabled
+          ExpirationInDays: '90'
+
+  CloudFront:
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Enabled: 'true'
+        Origins:
+        - DomainName: !Sub # Get website assets from S3 bucket
+          - ${WebsiteBucket}.${S3WebEndpoint}
+          - WebsiteBucket: !Ref WebsiteBucket
+            S3WebEndpoint: !FindInMap [RegionMap, !Ref "AWS::Region", WebsiteEndpoint]
+          Id: !Sub
+            - ${Subdomain}-${DomainName}
+            - Subdomain: !FindInMap [ StageDetails, !Ref Stage, Subdomain]
+          CustomOriginConfig:
+            OriginProtocolPolicy: http-only
+        Comment: !Sub
+          - Distribution for ${URL}
+          - URL: !Sub
+            - ${Subdomain}.${DomainName}
+            - Subdomain: !FindInMap [ StageDetails, !Ref Stage, Subdomain]
+        HttpVersion: http2
+        Logging:
+          IncludeCookies: 'false'
+          Bucket: !Sub ${LogBucket}.s3.amazonaws.com
+        Aliases: # Allowed URLs for distribution
+        - !Sub
+          - ${Subdomain}.${DomainName}
+          - Subdomain: !FindInMap [ StageDetails, !Ref Stage, Subdomain]
+        DefaultCacheBehavior:
+          AllowedMethods: # TODO: Should these all be allowed? Why not just GET & HEAD & OPTIONS
```

Comment:
GET, HEAD and OPTIONS should be all we need...

Reply:
Thanks, this used to have `POST`, `PUT`, and `DELETE` so I put the TODO, I ended up removing those but forgot to remove the TODO.